### PR TITLE
[AA-1239] - Display external API URLs under Docker

### PR DESCRIPTION
**Pull request by saa14**
**Description**

**NOTE: This PR is a companion of the  [Docker counterpart](https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS-Docker/pull/50)**

- Added ApiExternalUrl appsetting to set the ods/api url. This appsetting is only supposed to be used for displaying the correct url in docker environments.
- Added a private method GetApiUrlForDisplay to replace the production Api Url with the specified ApiExternalUrl if present. Refactored the controller actions to use the method.
- Updated the build script and BuildDockerDevelopment script to take the new appsetting of ApiExternalUrl into account.

### CODE REVIEW

File: BuildDockerDevelopment.ps1 [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 $p = @{
         ProductionApiUrl = "http://api"
+        ApiExternalUrl = "https://localhost:5001"
```
Comment (plioi) [inline BuildDockerDevelopment.ps1:21]: Please include similar example over in the documentation comment of build.ps1.

File: Application/EdFi.Ods.AdminApp.Web/Controllers/ApplicationController.cs [hunk 2]
```diff
@@ -214,19 +224,32 @@ public ActionResult RegenerateSecret(RegenerateSecretModel model)
...
+                ApiUrl = apiUrl
             });
         }
 
         private List<string> GetClaimSetNames()
         {
             return _getClaimSetNamesQuery.Execute().Except(CloudOdsAdminApp.SystemReservedClaimSets).ToList();
         }
+
+        private static string GetApiUrlForDisplay(string apiUrl)
```
Comment (plioi) [inline Application/EdFi.Ods.AdminApp.Web/Controllers/ApplicationController.cs:247]: I think this should be a get-only property on CloudOdsAdminAppSettings.Instance. A clue that it belongs there is that the body of this method asks that object for a few values and then calculates the result with no need for other inputs.

Comment (plioi) [reply Application/EdFi.Ods.AdminApp.Web/Controllers/ApplicationController.cs:247]: I also prefer not to need mutation of the input parameter here.

**Comment by stephenfuqua** [comment]
@saa14 please bump the version to 2.1.0 in `.teamcity\_Self\Project.kt`.

**Comment by plioi** [comment]
@stephenfuqua You mean to 2.1.1, correct? The current value is 2.1.0 set a few months back.

**Review by plioi — APPROVED** [review]

**Pull request merged**